core/internal/stream/handler.go

```diff
@@ -845,6 +856,18 @@ func (h *Handler) handleRequestGetSystemMetrics(record *spb.Record) {
 	h.respond(record, response)
 }
 
+func (h *Handler) handleRequestGetSystemMetadata(record *spb.Record) {
+	response := &spb.Response{}
+
+	response.ResponseType = &spb.Response_GetSystemMetadataResponse{
+		GetSystemMetadataResponse: &spb.GetSystemMetadataResponse{
+			Metadata: h.metadata,
+		},
+	}
```

Comment:
you can just merge it into one:
```suggestion
	response := &spb.Response{
		ResponseType: &spb.Response_GetSystemMetadataResponse{
			GetSystemMetadataResponse: &spb.GetSystemMetadataResponse{
				Metadata: h.metadata,
			},
		},
	}
```